### PR TITLE
Pip package for MMS 1.0 alpha

### PR DESCRIPTION
**Pull request by vdantu**
*Issue #, if available:*

*Description of changes:*
Pip package is available at test pypi for MMS 1.0

pip install -U --index-url https://test.pypi.org/simple/  --pre --extra-index-url https://pypi.org/simple mxnet-model-server --no-cache-dir

run 
mxnet-model-server -h for how to run the mms

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mms/model_service_worker.py [hunk 1]
```diff
@@ -37,10 +37,14 @@ class MXNetModelServiceWorker(object):
     """
     Backend worker to handle Model Server's python service code
     """
-    def __init__(self, s_name=None):
-        if s_name is None:
-            raise ValueError("Incomplete data provided: Model worker expects \"socket name\"")
-        self.sock_name = s_name
+    def __init__(self, s_type=None, s_name=None):
```
Comment (vdantu) [inline mms/model_service_worker.py]: Ignore this. Its from another PR.

File: mms/model_service_worker.py [hunk 2]
```diff
@@ -17,11 +17,11 @@
```
Comment (vdantu) [inline mms/model_service_worker.py]: Ignore this file for this PR

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java [hunk 3]
```diff
@@ -17,6 +17,8 @@
 import java.io.File;
```
Comment (vdantu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: Ignore this file for this PR. Its being tracked by another PR.

File: setup.py [hunk 4]
```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
...
+dest_file_name = os.path.abspath('mms/frontend/model-server.jar')
+
+
+def pypi_description():
+    """Imports the long description for the project page"""
+    with open('PyPiDescription.rst') as df:
+        return df.read()
+
+
+def detect_model_server_version():
```
Comment (frankfliu) [inline setup.py]: Version should be passed as argument during the build.
If version is not provided, then use default version

File: setup.py [hunk 5]
```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
...
+    with open(os.path.abspath(os.path.join("mms", "version.py")), 'r') as vf:
+        exec(vf.read(), None, globals())
+    return __version__
+
+
+def select_mxnet(requirements=[]):
```
Comment (frankfliu) [inline setup.py]: This should be passed as parameter as well, should not depend on current host.

Also, "requirements" shadows variable in outer scope. You don't really need pass it in, you can append the return value at calling function.

File: setup.py [hunk 6]
```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
...
+    if platform.system().lower() == 'linux':
+        try:
+            # Check if CUDA is installed
+            cuda = ctypes.cdll.LoadLibrary('libcudart.so')
+            device_count = ctypes.c_int()
+            # get the number of supported GpUs
+            cuda.cudaGetDeviceCount(ctypes.byref(device_count))
+            if device_count.value > 0:
+                gpu_platform = True
+        except Exception as e:
```
Comment (frankfliu) [inline setup.py]: Local variable e is not used

File: setup.py [hunk 7]
```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
...
+        name='mxnet-model-server',
+        version=version.strip()+'a2',
+        description='Model Server for Apache MXNet is a tool for serving neural net models for inference',
+        long_description=pypi_description(),
+        url='https://github.com/awslabs/mxnet-model-server',
+        keywords='MXNet Model Server Serving Deep Learning Inference AI',
+        packages=pkgs,
+        cmdclass={
+            'build_frontend': BuildFrontEnd,
+            'build_py':BuildPy,
```
Comment (frankfliu) [inline setup.py]: Missing space

File: setup.py [hunk 8]
```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
...
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        try:
+            front_end_bin_dir = os.path.abspath('.') + '/mms/frontend'
```
Comment (frankfliu) [inline setup.py]: os.path.abspath('.') might throw exception, and will cause crash in exception handling code, due to front_end_bin_dir has not been initialized.

File: setup.py (hunk 4 above)
Comment (vdantu) [reply setup.py]: Yes, we will need to look into this a little more in detail. I would like to take it up as a part of putting it together in the pipeline.

File: setup.py (hunk 5 above)
Comment (vdantu) [reply setup.py]: True. This can be removed.

File: setup.py (hunk 6 above)
Comment (vdantu) [reply setup.py]: True. This can be removed as well.

File: setup.py (hunk 7 above)
Comment (vdantu) [reply setup.py]: True added it. I should probably look into attaching a pylint for this file as well.

File: setup.py (hunk 8 above)
Comment (vdantu) [reply setup.py]: The assumption with setup.py is we run python build in the same directory where setup.py exists. If there is an exception, it is raised to the above level and the build for setup.py will fail.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java [hunk 9]
```diff
@@ -16,15 +16,16 @@
 import com.amazonaws.ml.mms.util.ConfigManager;
 import com.amazonaws.ml.mms.util.JsonUtils;
 import com.amazonaws.ml.mms.util.NettyUtils;
+<<<<<<< HEAD
```
Comment (piyushghai) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: Can you remoe the merge conflicts here ?

File: mms/model_service_worker.py [hunk 10]
```diff
@@ -21,7 +21,10 @@
 from builtins import bytes
 from builtins import str
 
+<<<<<<< HEAD
 from mms.model_loader import ModelLoader
+=======
```
Comment (piyushghai) [inline mms/model_service_worker.py]: Merge conflicts.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java (hunk 9 above)
Comment (piyushghai) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerLifeCycle.java]: Oh they are removed in a separate commit.  Just saw.

File: mms/model_service_worker.py (hunk 10 above)
Comment (piyushghai) [reply mms/model_service_worker.py]: Oh they are removed in a separate commit. Just saw.

**Pull request merged**